velox/exec/tests/utils/ExpressionBuilder.h

```diff
@@ -0,0 +1,311 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/parse/Expressions.h"
+
+namespace facebook::velox::expr::builder {
+
+/// Fluent Expression Builder.
+///
+/// This file contains fluent methods that make it convenient to create
+/// (untyped) expression trees. This provides similar functionality to a SQL
+/// parser, without briging dependency on external libraries or bringing leaked
+/// semantics from other systems.
+///
+/// The untyped expressions can then be turned into typed expressions ready for
+/// execution using type binding from `core::Expressions::inferTypes()`.
+///
+/// The API provided is as close to the actual expression trees as possible.
+/// Filters, arithmetics, function calls, literals, aliases, and more are
+/// supported with this API.
+///
+/// For example, to create a field reference (reference to a column), one can:
+///
+/// > using namespace expr::builder;
+/// > ExprPtr e = field("col");
+///
+/// Or use the "_f" literal provided:
+///
+/// > ExprPtr e = "col"_f;
+///
+/// To debug the expression generated, you can simply:
+///
+/// > LOG(INFO) << *e;
+///
+/// Filters (comparisons) can be fluently created using C++ overloaded
+/// operators
+///
+/// > ExprPtr e = "col"_f > 10; // "col > 10"
+/// > ExprPtr e = "col"_f != "bar"; // "col != 'bar'"
+///
+/// Null checking filters:
+///
+/// > ExprPtr e = isNull("col"_f) // "col is null"
+/// > ExprPtr e = !isNull("col"_f) // "col is not null"
+///
+/// Conjuncts and "between":
+///
+/// > ExprPtr e = "a"_f && "b"_f || "c"_f; // "a AND b OR c"
+/// > ExprPtr e = between("a"_f, 0L, 10L); // "a between 0 and 10"
+///
+/// C++ literals are automatically converted into ConstantExpr (expression
+/// literals) when part of an expression. To explicitly create a literal:
+///
+/// > ExprPtr e = literal(10.3);
+/// > ExprPtr e = literal("str");
+/// > ExprPtr e = literalTimestamp("1970-01-01 01:30:00");
+///
+/// Arithmetic operators are also overloaded:
+///
+/// > ExprPtr e = "col"_f + 1 - 10; // "col + 1 - 10"
+///
+/// In any expression, as long as one of the sides is an expression node, the
+/// correct expression will be created. For example, both version work as
+/// expected:
+///
+/// > ExprPtr e = "col"_f * 100; // "col * 100"
+/// > ExprPtr e = 100 * "col"_f; // "100 * col"
+///
+/// When building long expressions, be careful about C++ operator precedence.
+/// For example:
+///
+/// > ExprPtr e = "col"_f + 5 * 100;
+///
+/// will generate the expression "col + 500" since C++ will first sum the two
+/// integers. To force the expected behavior, you can explicitly spell out the
+/// literal:
+///
+/// > ExprPtr e = "col"_f + 5 * literal(100);
+/// > ExprPtr e = "col"_f + literal(5) * 100;
+///
+/// Both will generate "col + 5 * 100", which is "plus(col, multiply(5, 100))".
```

Comment:
Does this implicitly convert + to "plus" function call? Isn't this assuming "semantics"? Perhaps, list all such implicit conversions somewhere.